Add co-branded Jetpack & WooCommerce header during connection process

justinshreve opened the pull request:
This PR builds upon the WooCommerce styles added in #32993 by updating the header to contain a co-branded WooCommerce & Jetpack logo. The header is styled per the designs at p6riRB-4rG-p2 and woocommerce/woocommerce-admin#2152. It also updates the heading and sub-heading copy.

#### Testing instructions

To Test:

* Disconnect Jetpack `Jetpack > Dashboard > Connections > Site connection > Manage site connection > Disconnect`.
* Click “Setup Jetpack”.
* On the connection/authorization screen ( https://wordpress.com/jetpack/connect/authorize?client_id= ….. ) copy and paste the whole URL.
* Replace `https://wordpress.com` with ` http://calypso.localhost:3000` and replace the `from` parameter in the URL (it may be something like `connection-banner` or `landing-page-bottom`) with  `from=woocommerce-setup-wizard` . See Pastebin `213d5`  or ping me if you have issues.
* Hit enter, and you should see the new logo and copy.
* Disconnect
* Logout
* Try the same same URL logged out
* Test the standard Jetpack connection process, without the WooCommerce query string appended, and verify you see the standard Jetpack Onboarding styles, logo, and copy.


<img width="717" alt="Screen Shot 2019-05-20 at 2 56 57 PM" src="https://user-images.githubusercontent.com/689165/58045253-ea787700-7b0f-11e9-8997-e9d9fb905234.png">

<img width="789" alt="Screen Shot 2019-05-20 at 2 57 14 PM" src="https://user-images.githubusercontent.com/689165/58045254-ea787700-7b0f-11e9-85c0-c3ac8caf3459.png">

simison commented on client/blocks/login/style.scss:
```diff
@@ -9,6 +9,60 @@
+.layout.is-jetpack-woocommerce-flow {
+	.login__jetpack-logo {
+		text-align: center;
+		border-bottom: 1px solid #1f112e;
```
Could this hardcoded hex value be switched to CSS variable?

justinshreve replied:
Done in https://github.com/Automattic/wp-calypso/pull/33073/commits/ec4b2b0da4a0c1137a0b4760fb87999cff3137aa.

simison commented on client/components/jetpack-header/woocommerce.jsx:
```diff
@@ -0,0 +1,72 @@
+		<path
+			d="M23.7,0.2 L222.8,0.2 C235.4,0.2 245.6,10.4 245.6,23 L245.6,99 C245.6,111.6 235.4,121.8 222.8,121.8 L151.4,121.8 L161.2,145.8 L118.1,121.8 L23.8,121.8 C11.2,121.8 1,111.6 1,99 L1,23 C0.9,10.5 11.1,0.2 23.7,0.2 Z"
+			id="Shape"
+			fill="#342148"
```
Is this fill overwritten with CSS later and how does it play with the dark color theme in Calypso?

justinshreve replied:
`lightColorSchemeLogo` doesn't actually get used. `darkColorSchemeLogo` does, and I do not overwrite the colors with CSS.

I just wanted to provide a version for completeness as all of the other co-branded logos have alternatives.

simison replied:
Sounds good!

simison commented on client/jetpack-connect/style.scss:
```diff
@@ -882,6 +882,57 @@ $colophon-height: 50px; // wpcomColophon element at the bottom
+
+		.jetpack-connect__main-logo {
+			text-align: center;
+			border-bottom: 1px solid #1f112e;
```
Hardcoded hex, see https://github.com/Automattic/wp-calypso/pull/33073/files#r284687907

justinshreve replied:
Done in https://github.com/Automattic/wp-calypso/pull/33073/commits/ec4b2b0da4a0c1137a0b4760fb87999cff3137aa.

simison commented on client/login/wp-login/style.scss:
```diff
@@ -186,7 +186,12 @@ $image-height: 47px;
+
 	.wp-login__links a {
 		border-bottom: none;
+		line-height: 2.5em;
```
These `line-height` and `margin-top` seem _different_ from just changing colours like the other `is-jetpack-login`/`is-jetpack-woocommerce-flow` stuff seems to be. Should they be applied to all variations of the login page instead?

I didn't have a look how they actually change the layout and I appreciate if they're brand specific of course. :-) Just presense of these might surprise folks doing changes in generic layout.

justinshreve replied:
I added this just to tighten up the footer a bit, because it felt like there was too much spacing without the border separating it like the normal Calypso flow does. It didn't come from our design/brand guidelines, but the area looked off to me.

Before:

<img width="426" alt="Screen Shot 2019-05-16 at 9 05 05 AM" src="https://user-images.githubusercontent.com/689165/57855889-bb819e80-77b9-11e9-9d54-6b8b9929ca11.png">

After:

<img width="469" alt="Screen Shot 2019-05-16 at 9 03 57 AM" src="https://user-images.githubusercontent.com/689165/57855898-c3414300-77b9-11e9-9178-7daaf5b7c95e.png">

simison replied:
Oh right, makes sense!

I don't know if more generic way to implement that would be more clarity, or maybe just an inline code comment would suffice? This totally isn't blocking comment, feel free to ship without any changes. I'm just noting what seemed confusing to me.

I think the idea of 4em line-height was that it follows material patterns of 56px space for "a single line list item that has an icon".

https://material.io/design/components/lists.html#specs

Seems like 48px would be good as well if there's no icon but no idea how that translates to `em` value. We have this now but I don't know how to use it:

https://github.com/Automattic/wp-calypso/blob/641ebd2a0061877ca281dd52e59d7ac9f0a1ce76/packages/muriel-style/typography.scss

cc @griffbrad @blowery :wave: might know.

...maybe this list view isn't following Material/Muriel patterns at all and this isn't relevant. :-)

justinshreve replied:
Good call - we do want Muriel patterns. I hadn't seen anything in the Muriel guide lines about line-height, so thanks for the reminder about looking at Material :).

I went ahead and adjusted this to 3.4em (which seems like the correct conversion at font-size 14) for now. While not as big as a difference, it's still looking better to me.

We also plan on doing a design sweep of these PRs in context with the whole onboarding flow (part of which will be self hosted for the WooCommerce steps), so we can make more updates in the future if needed.

simison commented on client/jetpack-connect/main-wrapper.jsx:
```diff
@@ -40,10 +40,13 @@ export class JetpackConnectMainWrapper extends PureComponent {
 			'is-mobile-app-flow': !! retrieveMobileRedirect(),
 		} );
 
+		const partnerSlug = isWoo ? 'woocommerce' : this.props.partnerSlug;
```
I felt like sticking to data flow where you'd use either only `isWoo=true` _or_ `parterSlug='woocommerce` would be more clear and less surprising. That said, I understand this might be a constraint you're having with `JetpackHeader` that was introduced in the codebase already before this PR.

This definitely isn't blocking comment but are there cons on using `partnerSlug` across all these components?

justinshreve replied:
I can't think of any negative side-effects code wise.

Though one of the reasons I went with the `isWoo` pattern was from the existing separation I already saw here: https://github.com/Automattic/wp-calypso/blob/9b32491717f9f76a37c8fbc26572f22f6ccaf294/client/jetpack-connect/authorize.js#L230-L235, and it felt more future proof to me. There have been discussions about making it possible for WooCommerce / WooCommerce Services to just bundle authentication/sync modules and not rely on the full Jetpack plugin. It feels more right to consider this a different 'flow' to me than just one of the branded partner flows, since some of that branding could actually be dropped in the future. It also doesn't feel like WooCommerce is "partner" in the same way the hosts are, though I know that may be a bit pedantic.

justinshreve replied:
Another consideration is the opportunity for us in the future to co-brand this version of onboarding as well (I don't believe we have plans for this yet, but it's been raised). Having both properties would be useful -- so I can actually continue the `isWoo` pattern in `JetpackHeader` too.

justinshreve replied:
In https://github.com/Automattic/wp-calypso/pull/33073/commits/ec4b2b0da4a0c1137a0b4760fb87999cff3137aa, I implemented a `isWoo` parameter to keep this consistent and for the reasons above. I also added a `@todo` for handling mixing of `isWoo` and `partnerSlug` for further co-branded flows, but this is not currently in the scope of the project.

simison replied:
Sounds great and totally valid reasoning :thumbsup: Thanks for giving it such an in-depth consideration!

justinshreve commented:
Thank you for the great feedback! I've updated that color to a CSS variable, adjusted the line-height on those login links, and persisted the `isWoo` data flow into the `JetpackHeader` component.

timmyc commented on client/components/jetpack-header/index.jsx:
```diff
@@ -23,11 +23,29 @@ export class JetpackHeader extends PureComponent {
-		const { darkColorScheme, partnerSlug, width } = this.props;
+		const { darkColorScheme, partnerSlug, width, isWoo } = this.props;
+
+		if ( isWoo ) {
```
Minor but I suppose one route here would be to add a conditional for setting `partnerSlug` to `woocommerce` based on if `isWoo` is truthy or not... then instead of having this conditional return block, we could just add another case statement below

justinshreve replied:
I originally went that route, but see the discussion here: https://github.com/Automattic/wp-calypso/pull/33073#discussion_r284695767

timmyc replied:
Aye, thanks for the background!

MichaelArestad commented:
I think we should use the same design language across our cobranded login screens. Here's the one from Pressable:

![image](https://user-images.githubusercontent.com/1123119/57881589-ed592c00-77de-11e9-8b91-d9a1f175dbeb.png)

The biggest difference is the removal of the black line under the logos and the larger, brighter logos. I recommend going the same route as the Pressable cobranding by making the logos bigger and brighter. I also think the dark line (probably a bottom border) can be removed. What do you think?

MichaelArestad commented:
Also, I tested this, and it worked as expected. 👍

timmyc commented:
/cc @jameskoster for

> The biggest difference is the removal of the black line under the logos and the larger, brighter logos. I recommend going the same route as the Pressable cobranding by making the logos bigger and brighter. I also think the dark line (probably a bottom border) can be removed. What do you think?

jameskoster commented:
This design is based on the recent Muriel sprint where we sought to align account steps across our products. It carries through to the rest of our on-boarding experience. So if this step looked / felt different to the rest, that would create unwanted friction and disruption.

Based on this, if anything, I would suggest that the other co-branded screens are updated to match this one. But that deserves some discussion as we didn't definitively lock those designs down.

For now, if it's not a lot of work to make this particular screen match the original designs (seems like it's already done) then I think we should stick with it to avoid the aforementioned friction.

flootr commented on client/blocks/login/style.scss:
```diff
@@ -9,6 +9,57 @@
+.layout.is-jetpack-woocommerce-flow {
+	.login__jetpack-logo {
+		text-align: center;
+		border-bottom: 1px solid $muriel-gray-50;
```
Hey! Out of curiosity, is there a reason why you’re using a sass var in this case instead of css custom props?

justinshreve replied:
No reason -- thank you for catching that. I've fixed it up.

flootr replied:
Thanks! :+1:
